### PR TITLE
FAI-668: Explainability: Native build is broken

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.redhat.com/browse/FAI-668

See https://github.com/kiegroup/kogito-apps/pull/1110 specifically [this](https://github.com/kiegroup/kogito-apps/pull/1110#issuecomment-986684639).

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

**Comment by manstis** [comment]
Jenkins run native

**Comment by manstis** [comment]
Jenkins run native

**Comment by manstis** [comment]
Jenkins run native

**Comment by manstis** [comment]
Jenkins run native

**Comment by manstis** [comment]
Jenkins run native

**Comment by manstis** [comment]
Jenkins run native

**Comment by manstis** [comment]
Jenkins run native

**Comment by manstis** [comment]
Jenkins run native

File: apps-integration-tests/integration-tests-trusty-service/integration-tests-trusty-service-common/src/main/java/org/kie/kogito/it/trusty/AbstractTrustyExplainabilityEnd2EndIT.java [hunk 1]
```diff
@@ -211,34 +211,41 @@ public void doTest() {
...
+                    .untilAsserted(() -> {
+                        executionIds.forEach(executionId -> {
                             SalienciesResponse salienciesResponse = given()
                                     .port(trustyService.getFirstMappedPort())
                                     .auth().oauth2(accessToken)
                                     .when().get("/executions/decisions/" + executionId + "/explanations/saliencies")
                                     .then().statusCode(200)
                                     .extract().as(SalienciesResponse.class);
 
+                            System.out.printf("LIME response: status=%s, details=%s%n",
```
Comment (r00ta) [inline apps-integration-tests/integration-tests-trusty-service/integration-tests-trusty-service-common/src/main/java/org/kie/kogito/it/trusty/AbstractTrustyExplainabilityEnd2EndIT.java]: logger?

File: explainability/explainability-core/pom.xml [hunk 2]
```diff
@@ -34,7 +34,7 @@
     </dependency>
     <dependency>
       <groupId>org.optaplanner</groupId>
-      <artifactId>optaplanner-core</artifactId>
+      <artifactId>optaplanner-quarkus</artifactId>
```
Comment (r00ta) [inline explainability/explainability-core/pom.xml:37]: just for my understanding: why this change?

**Review by r00ta — APPROVED** [review]

File: apps-integration-tests/integration-tests-trusty-service/integration-tests-trusty-service-common/src/main/java/org/kie/kogito/it/trusty/AbstractTrustyExplainabilityEnd2EndIT.java (hunk 1 above)
Comment (manstis) [reply apps-integration-tests/integration-tests-trusty-service/integration-tests-trusty-service-common/src/main/java/org/kie/kogito/it/trusty/AbstractTrustyExplainabilityEnd2EndIT.java]: Nooooooooo. Delete. This is left over from me trying to debug why it was failing.

File: explainability/explainability-core/pom.xml (hunk 2 above)
Comment (manstis) [reply explainability/explainability-core/pom.xml:37]: Because @triceo told me to change it if we're running with Quarkus (to avoid the nasty Quarkus errors relating to classes being on the heap when building images due OptaPlanner's use of Drools)..

**Comment by manstis** [comment]
Jenkins run native

File: explainability/explainability-core/pom.xml (hunk 2 above)
Comment (triceo) [reply explainability/explainability-core/pom.xml:37]: There is a distinction here:
- If you have a module X which is not using Quarkus, continue using `optaplanner-core` in module X.
- However, if module Y uses module X, and module Y uses Quarkus, then module Y needs to exclude `optaplanner-core` and bring in `optaplanner-quarkus` instead.
 
This will ensure that everything runs properly both in Quarkus and outside.
